enhancements/monitoring/client-cert-scraping.md

```diff
@@ -0,0 +1,132 @@
+is unavailable or unreachable.
+To do this, we will combine a local authorizer (already merged in many binaries and the rbac-proxy) and client-cert based
+authentication to have a fully local authentication and authorization path for scraper targets.
+
+## Motivation
+
+If networking (or part of networking) is down and a scraper target cannot reach the kube-apiserver to verify a token
+and a subjectaccessreview, then the metrics scraper can be rejected.
+The subjectaccessreview (authorization) is already largely addressed, but service account tokens are still used
+for scraping targets.
+Tokens require an external network call that we can avoid by using client certificates.
+Gathering metrics, especially client metrics, from partially functionally clusters helps narrow the search area between
+kube-apiserver, etcd, kubelet, and SDN considerably.
+
+In addition, this will significantly reduce the load on the kube-apiserver.
+We have observed in the CI cluster that token and subject access reviews are a significant percentage of all kube-apiserver traffic.
+
+### Goals
+
+1. Gather all possible metrics even when targets cannot reach the kube-apiserver.
+2. Avoid sending valid kube-apiserver credentials to everyone who can create a servicemonitor.
+
+### Non-Goals
+
+1. Force all servicemonitor targets to change at the same time.
+
+## Proposal
+
+1. The monitoriting operator creates a client cert/key pair valid against the kube-apiserver.
+   There is a CSR signer name specifically for this.
+2. The kube-controller-manager operator runs an approver that allows monitoring operator SA to get signed certificate for
+   The existing scraper identity.
+3. The monitoring operator puts the client cert/key pair into a secret.
+4. The monitoring operator keeps these credentials up to date (the signer is only good for 30 days or so).
+5. The metrics scraper mounts the client cert/key pair and ensures that it is hot-reloadable or builds a process suicider.
+   There is library-go code to suicide on file changes.
+6. A participating metrics scraper target, can read the in-cluster client cert CA bundle and create an authenticator
+   using it.
+   The default delegated authentication stack does this and on openshift, the configmap is exposed to system:authenticated.
+7. A servicemonitor grows a way to specify using this client cert.
+   Technically this is optional.
+   If you find it too difficult, just *always* use the client-cert for scraping since the secrets never leave disk.
+   Those targets which don't support it will simply ignore the client-cert.
+8. Profit.
+   Now we can collect metrics for at least
+   1. openshift-apiserver
+   2. openshift-controller-manager
+   3. kube-scheduler
+   4. all kubelets
+   5. node-exporter
+   6. all control plane operators
+   7. kube-controller-manager
+   8. etcd
+   These are the ones I've updated already and we'll be able to get more.
```

Comment:
All monitoring components (except Prometheus, Alertmanager and Thanos querier) also use kube-rbac-proxy to secure their metrics endpoint. The 3 others got scraped via oauth-proxy (because historical reasons I guess).